stock_picking_batch_extended/views/stock_move_views.xml

```diff
@@ -0,0 +1,68 @@
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <record id="view_move_line_tree" model="ir.ui.view">
+        <field name="model">stock.move.line</field>
+        <field name="inherit_id" ref="stock_picking_batch.view_move_line_tree" />
+        <field name="mode">primary</field>
+        <field name="priority">999</field>
+        <field name="arch" type="xml">
+            <field name="product_id" position="before">
+                <field name="picking_code" column_invisible="True" />
+                <field name="is_locked" column_invisible="True" />
+                <field name="lots_visible" column_invisible="True" />
+            </field>
+            <field name="picking_id" position="attributes">
+                <attribute name="domain" />
+                <attribute name="optional">show</attribute>
+            </field>
+            <field name="lot_id" position="attributes">
+                <attribute
+                    name="readonly"
+                >state in ('done', 'cancel') or is_locked</attribute>
+                <attribute name="invisible">not lots_visible</attribute>
+                <attribute name="optional">show</attribute>
+            </field>
+            <field name="lot_name" position="attributes">
+                <attribute
+                    name="readonly"
+                >state in ('done', 'cancel') or is_locked</attribute>
+                <attribute name="invisible">not lots_visible</attribute>
+                <attribute name="optional">hide</attribute>
+            </field>
+            <field name="location_id" position="attributes">
+                <attribute
+                    name="readonly"
+                >picking_code == 'incoming' or (state in ('done', 'cancel') and is_locked)</attribute>
+                <attribute name="optional">show</attribute>
+            </field>
+            <field name="location_dest_id" position="attributes">
+                <attribute
+                    name="readonly"
+                >picking_code == 'outgoing' or (state in ('done', 'cancel') and is_locked)</attribute>
+                <attribute name="optional">show</attribute>
+            </field>
+            <field name="package_id" position="attributes">
+                <attribute
+                    name="readonly"
+                >state in ('done', 'cancel') or is_locked</attribute>
+                <attribute name="optional">show</attribute>
+            </field>
+            <field name="result_package_id" position="attributes">
+                <attribute
+                    name="readonly"
+                >state in ('done', 'cancel') or is_locked</attribute>
+                <attribute name="optional">show</attribute>
+            </field>
+            <field name="quantity" position="attributes">
+                <attribute
+                    name="invisible"
+                >state in ('done', 'cancel') or is_locked</attribute>
```

Comment:
These conditions make it impossible to modify the field...
In 15.0 this was the condition:
https://github.com/OCA/stock-logistics-workflow/blob/15.0/stock_picking_batch_extended/views/stock_move_views.xml#L66

However, what really needs to be replicated are the conditions of the lines in the delivery note.